src/Proto3/Suite/DotProto/Generate.hs

```diff
@@ -472,13 +537,50 @@ dpptToHsType ctxt = \case
   Bool     -> pure $ primType_ "Bool"
   Float    -> pure $ primType_ "Float"
   Double   -> pure $ primType_ "Double"
+  {-| For properly handling wrapper values found at:
+      <https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/wrappers.proto>
+      This functionality is new for now and hides behind the --unwrap flag.
+  -}
+  Named (Dots (Path ("google" :| ["protobuf", x])))
+    | x == "Int32Value"  -> pure $ primType_ "Int32"
```

Comment:
Instead of creating two separate `dppToHsTypeOld` and `dppToHsType` functions, you can just add a check on each of these branches to make sure that legacy mode is not enabled.  That way you don't have to worry about code drift between the legacy and non-legacy code

Reply:
@Gabriel439 Got it! I will do this.